client/my-sites/stats/activity-log-tasklist/index.jsx

```diff
@@ -0,0 +1,290 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { isEmpty, get, merge, each, omit, keyBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ActivityLogTaskUpdate from './update';
+import alPluginsToUpdate from './plugins-to-update';
+import Card from 'components/card';
+import PopoverMenuItem from 'components/popover/menu-item';
+import EllipsisMenu from 'components/ellipsis-menu';
+import Gridicon from 'gridicons';
+import { getSite } from 'state/sites/selectors';
+import { updatePlugin } from 'state/plugins/installed/actions';
+import { getStatusForPlugin } from 'state/plugins/installed/selectors';
+import PluginNotices from 'lib/plugins/notices';
+import { errorNotice, infoNotice, successNotice, removeNotice } from 'state/notices/actions';
+
+/**
+ * Checks if the supplied plugin is currently updating.
+ *
+ * @param {object} plugin Plugin to check status
+ * @returns {bool} True if plugin is updating.
+ */
+const isPluginUpdating = plugin => 'inProgress' === plugin.updateStatus.status;
+
+class ActivityLogTasklist extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		pluginsWithUpdates: PropTypes.arrayOf( PropTypes.object ), // Plugins with pending updates
+		allPluginsToUpdate: PropTypes.arrayOf( PropTypes.object ), // Don't use. Use connected pluginsToUpdate instead.
```

Comment:
this is a bit concerning. if we aren't supposed to use it, why are we specifying it?

Reply:
This is a prop passed by the wrapper. I wanted to declare it to have the type checking. This is only used in the `connect`. I'll update the comment since anybody can actually use this if they want to.

Reply:
If it's for `connect` or the wrapper then we should remove it from the wrapped. That's introducing backwards coupling that will confuse people.

We can either ignore it here or explicitly lift it out of the passed attributes from `connect()`.

If you want the type-checking inside of `connect()` then we can expand the connected component into its own component and add explicit `propTypes` to it.

```js
const innerComponent = connect( … )( Component );
innerComponent.propTypes = { … }

export default innerComponent;
```